.github/actions/validate-jargon-artefacts/src/contextValidation.js

```diff
@@ -29,16 +30,42 @@ async function validateContext(jsonldContext) {
 /**
  * Validate JSON-LD context in credentials
  * @param rawArtefactData is an array of JSON schemas and instances
+ * @param jsonldContext is the context object provided by Jargon
  * @returns { valid: boolean }
  */
-async function validateContextInCredential(rawArtefactData) {
+async function validateContextInCredential(rawArtefactData, jsonldContext) {
   const { instances } = splitSchemasAndInstances(rawArtefactData);
   const instanceFileNames = Object.keys(instances);
 
-  // Fetch all instances in parallel
-  const fetchedInstances = await Promise.all(instanceFileNames.map(async instanceFileName => {
+  // Filter out secondary instances that lack @context.
+  const mainInstanceFileNames = instanceFileNames.filter(instanceFileName =>
+    JARGON_INSTANCES_TO_VALIDATE.some(validInstance => instanceFileName.includes(validInstance))
+  );
+
+  if (!mainInstanceFileNames.length) {
+    core.info(`
+      No permitted sample instances found for JSON-LD expansion. 
+      Ensure the instance file name is present in JARGON_INSTANCES_TO_VALIDATE. 
+      Current JARGON_INSTANCES_TO_VALIDATE: ${JARGON_INSTANCES_TO_VALIDATE.join(', ')}
+    `);
+    return { valid: false };
+  }
+
+  // Fetch all permitted instances in parallel
+  const fetchedInstances = await Promise.all(mainInstanceFileNames.map(async instanceFileName => {
     const instanceJson = await fetchArtefactData(instances[instanceFileName]);
 
+    if (instanceJson && jsonldContext && JARGON_CONTEXT_IRI_PREFIX) {
+      const context = instanceJson['@context'];
+
+      if(context && Array.isArray(context) && jsonldContext.url) {
+      // Replace the deployed @context IRI with the Jargon context IRI
+      instanceJson['@context'] = context.map(context =>
+          context.includes(JARGON_CONTEXT_IRI_PREFIX) ? jsonldContext.url : context
```

Comment:
I wouldn't shadow the main variable (it confuses reading the code), also `startsWith` is more appropriate than `includes` here?

```suggestion
      instanceJson['@context'] = context.map(c =>
          c.startsWith(JARGON_CONTEXT_IRI_PREFIX) ? jsonldContext.url : c
```

Also, do we need to be more careful here? JARGON_CONTEXT_IRI_PREFIX is the same for all the document types, so if a context array included two different contexts that both begin with that prefix (we don't have a reason for this currently, I know), then they'd both be replaced with the same jargon URL. See if you think it's worth testing for equality (a bit of extra work). Up to you.

Reply:
Absolutely. From my experience experimenting with this, I believe Jargon will need to expose additional data to achieve this. However, this could be due to the payloads I've been receiving.

The primary omission is the version value. In the payloads I've reviewed, version is set to working, likely because they are snapshots.

Further testing is required, but I've prioritised getting it working for this iteration :)